File: ReactNative/src/modules/signin/services/signin.service.ts
import axios from 'axios';
import { ISigninForm } from '../../../@types/signin-form';
import { ITokens } from '../../../@types/tokens';
import { setAllTokens } from '../../common/storage/key-chain';
import { SIGNIN_URL } from '../../common/utils/url';

export async function requestSignin(form: ISigninForm): Promise<void> {
  const tokens: ITokens = (await axios.post(SIGNIN_URL, form)).data;
  await setAllTokens(tokens);
}

File: flutter_todo/lib/src/modules/signin/services/signin_service.dart
import 'package:dio/dio.dart';
import 'package:flutter_todo/src/modules/common/storage/secure_storage.dart';
import 'package:flutter_todo/src/modules/common/utils/url.dart';
import 'package:flutter_todo/src/modules/signin/models/signin_model.dart';

Future<void> requestSignin(SigninModel formData) async => await Dio()
    .post(signinUrl, data: formData.toJson())
    .then((response) => setAllTokens(
        response.data['accessToken'], response.data['refreshToken']))
    .catchError((error) => throw error);
